src/resource-extensions/resource-extensions.js

```diff
@@ -132,21 +132,25 @@
 
             /* jshint -W003 */
             var methodResult = oldMethod.apply( this, methodargs );
+            var previousRequery = undefined;
 
             if ( action === 'query' ) {
               methodResult.requery = function() {
+                if ( previousRequery && previousRequery.$cancelRequest ) {

```

Comment:
Can we check if it needs cancelling? I'm concerned what might happen if we cancel something that's fulfilled and then user code checks that. It might get conflicting information

Reply:
Reading through the details for $resource and [cancelRequest](https://docs.angularjs.org/api/ngResource/service/$resource)
_Calling $cancelRequest() for a non-cancellable or an already completed/cancelled request will have no effect._. 
So successfully fulfilled requests will behave as expected. There isn't any in-built API to query if a request was cancelled, so the user code shouldn't be able to check if we even tried to cancel a request - regardless if whether cancelling actually anything. 
We could possibly do something like `methodResult.$cancelRequest !== angular.noop` to see if calling $cancelRequest would actually do anything - but smells a bit to me.

Reply:
> _Calling $cancelRequest() for a non-cancellable or an already completed/cancelled request will have no effect._

Thanks for digging up that info, Terry (I probably should have done that myself really 😞).

That satisfies my concerns.